thorlcr/thorutil/thormisc.hpp

```diff
@@ -359,6 +359,54 @@ class graph_decl CStreamFileOwner : public CSimpleInterfaceOf<IExtRowStream>
     }
 };
 
+// Tracks the current and peak storage used for some files
+class FilesSizeTracker: public CInterface
+{
+    RelaxedAtomic<unsigned __int64> activeSize{0};
+    RelaxedAtomic<unsigned __int64> peakSize{0};
+public:
+    void growSize(unsigned __int64 size)
+    {
+        if (size)
+        {
+            unsigned __int64 newActiveSize = activeSize.add_fetch(size);
+            peakSize.store_max(newActiveSize);
+        }
+    }
+    void shrinkSize(unsigned __int64 size)
+    {
+        if (size)
+            activeSize.fetch_sub(size);
+    }
+    unsigned __int64 queryActiveSize() const
+    {
+        return activeSize.load();
+    }
+    unsigned __int64 queryPeakSize() const
+    {
+        return peakSize.load();
+    }
+};
+
+class CFileOwnerSizeUpdater : public CFileOwner
```

Comment:
maybe not now, but I suspect every CFileOwner should have this functionality?
suggesting that code should be rolled into CFileOwner